modules/sdk-core/src/bitgo/tss/common.ts

```diff
@@ -30,24 +30,40 @@ export async function getTxRequest(bitgo: BitGoBase, walletId: string, txRequest
  * @param {String} walletId - the wallet id  *
  * @param {String} txRequestId - the txRequest Id
  * @param {SignatureShareRecord} signatureShare - a Signature Share
+ * @param requestType - The type of request being submitted (either tx or message for signing)
+ * @param signerShare
+ * @param mpcAlgorithm
+ * @param apiMode
  * @returns {Promise<SignatureShareRecord>} - a Signature Share
  */
 export async function sendSignatureShare(
   bitgo: BitGoBase,
   walletId: string,
   txRequestId: string,
   signatureShare: SignatureShareRecord,
+  requestType: RequestType,
   signerShare?: string,
   mpcAlgorithm: 'eddsa' | 'ecdsa' = 'eddsa',
   apiMode: 'full' | 'lite' = 'lite'
 ): Promise<SignatureShareRecord> {
-  let transactions = '';
-  if (mpcAlgorithm === 'ecdsa' || apiMode === 'full') {
-    transactions = '/transactions/0';
+  let addendum = '';
+  let urlPath;
+  switch (requestType) {
+    case RequestType.tx:
+      if (mpcAlgorithm === 'ecdsa' || apiMode === 'full') {
+        addendum = '/transactions/0';
+      }
+      urlPath = '/wallet/' + walletId + '/txrequests/' + txRequestId + addendum + '/signatureshares';
```

Comment:
You could move this line out of the switch statement, then you only need it once. It looks like you introduced the `addendum` variable so that you could do that.